packages/server/src/validators/HeaderValidator.ts

```diff
@@ -0,0 +1,107 @@
+import {TUS_VERSION, TUS_RESUMABLE} from '../constants'
+import {Metadata} from '../models'
+
+type validator = (value?: string) => boolean
+
+export const validators = new Map<string, validator>([
+  [
+    // The Upload-Offset request and response header indicates a byte offset within a resource.
+    // The value MUST be a non-negative integer.
+    'upload-offset',
+    function (value) {
+      return Number.parseInt(value as string, 10) >= 0
+    },
+  ],
+  [
+    // The Upload-Length request and response header indicates the size of the entire upload in bytes.
+    // The value MUST be a non-negative integer.
+    'upload-length',
+    function (value) {
+      return Number.parseInt(value as string, 10) > 0
```

Comment:
Also allow 0.

```suggestion
      return Number.parseInt(value as string, 10) >= 0
```

Reply:
I don't think `Number.parseInt` is appropriate for any validation. It parses many non-numerical inputs as valid numbers. Maybe simple regex?
```
const regex = /^[1-9][0-9]*$/
return value === '0' || regex.test(value);
```

Example of invalid input that would be parsed as valid number:
`"0100" -> 100`
`"0asd100" -> 100`
`"1asd100" -> 1`